sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala

```diff
@@ -63,16 +60,22 @@ private[hive] class SparkSQLSessionManager(hiveServer: HiveServer2, sqlContext:
     } else {
       sqlContext.newSession()
     }
+    sqlContext.sparkContext.setLocalProperty(
+      SQLContext.SESSION_ID_KEY, sessionHandle.getSessionId.toString)
```

Comment:
I think setting it in a thread local in open session is not enough, as I think that each request in the session can be handled by different threads of the connection threadpool.